packages/inferno-router/src/Route.ts

```diff
@@ -87,18 +120,36 @@ class Route extends Component<Partial<IRouteProps>, RouteState> {
         '<Route> elements should not change from controlled to uncontrolled (or vice versa). You provided a "location" prop initially but omitted it on a subsequent render.'
       );
     }
+    const match = this.computeMatch(nextProps, nextContext.router);
+    // Am I a match? In which case check for loader
+
+    if (nextProps?.loader) {
+      const params = match.params;
+      const request = undefined;
+      this.runLoader(nextProps.loader, params, request, match);
+      return;
+    }
 
-    this.setState({
-      match: this.computeMatch(nextProps, nextContext.router)
-    });
+    this.setState({ match });
   }
 
   public render() {
-    const { match } = this.state!;
-    const { children, component, render } = this.props;
+    const { match, __loaderData__ } = this.state!;
+    
+    // QUESTION: Is there a better way to invoke this on/after first render?
```

Comment:
Why this has to be in render method?

Reply:
It shouldn't be there... I have now moved it to `componentDidMount` where it belongs 😎